### PR TITLE
I added the string parser javascript and the paired html test file.

### PR DESCRIPTION
**Pull request by tmaher1**

### CODE REVIEW

File: src/js/model/string/string.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+var AMPParse = AMPParse || {};
+
+AMPParse.buildString = function (hexadecimal) {
+    nibblesUsed = 0; //int
+    nullTerminate = false; //bool
+    returnString = ""; //string
+    tempChars = "00";
+    
+    // Defensive programming
+    if (typeof hexadecimal === "undefined" || typeof hexadecimal !== "string") {
```
Comment (emanguy) [inline src/js/model/string/string.js:398]: You can probably get rid of the "undefined" test. We just have to verify that it's not a string to throw the reference error.

File: src/js/model/string/string.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	throw new ReferenceError("Provided parameter does not exist or is not a string");
+    }
+    tempChars = hexadecimal.substring(0, 2);
+    if (tempChars == "00") {
```
Comment (emanguy) [inline src/js/model/string/string.js:402]: If it immediately null terminates wouldn't it just be an empty string?

File: src/js/model/string/string.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	throw new RangeError ("The provided hex is too short to contain a string");
+    }
+
+    // Find length of string
+    while (!nullTerminate) {
+	tempChars = hexadecimal.substring(nibblesUsed, nibblesUsed+2);
+	if (tempChars != "00") {
+ 	    nibblesUsed++;
+	    nibblesUsed++;
+	    returnString += tempChars;
```
Comment (emanguy) [inline src/js/model/string/string.js]: You'll need to convert those bytes that you're pulling into actual characters (i.e. 0x41 = 'A')

File: src/js/model/string/test-string-model.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+	suiteSetup(function()
+	{
+		stringhex = "1ABCDE00FGHI";
+		returnedObj = AMPParse.buildString(stringhex);
+	});
+
+	test("returns the correct value", function()
+	{
+		expect(returnedObj.returnValue.value).to.be.a("string").and.equal("1ABCDE");
```
Comment (emanguy) [inline src/js/model/string/test-string-model.html]: The string you have here, 1A, BC, DE should be the string [SUB], (not in ascii), (not in ascii). Remember, each byte value gets converted into an ASCII character.

**Review by emanguy — CHANGES_REQUESTED** [review]
You're just outputting the raw hex string right now. It needs to get converted into ASCII values.

**Comment by tmaher1** [comment]
Closing, I'll make a new one with the fixed stuff.

**Pull request closed without merging**